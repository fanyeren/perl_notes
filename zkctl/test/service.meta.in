#!/bin/bash
source $(dirname $0)/common
TYPE=service
VALUE=bs_dnews_3
raw='"last_sync_time":"201202131117","lowest_ratio_of_up_service_instances":"66"'

node_clear $TYPE

./zkctl -w $TYPE add --$TYPE-id=$VALUE --data='{'$raw'}'
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
node_clear $TYPE
./zkctl -w $TYPE add --$TYPE-id=$VALUE --data={}
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE add-meta-attr --$TYPE-id=$VALUE --key=${kv[0]} --value=${kv[1]}
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE get-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got all $TYPE result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE del-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got empty $TYPE result result=$?"

./zkctl -w $TYPE del --$TYPE-id=$VALUE

node_clear $TYPE
