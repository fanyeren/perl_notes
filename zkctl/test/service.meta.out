{"last_sync_time":"201202131117","lowest_ratio_of_up_service_instances":"66"}
{}
Add meta last_sync_time=201202131117 result 0
201202131117
Get meta last_sync_time=201202131117 result 0
Add meta lowest_ratio_of_up_service_instances=66 result 0
66
Get meta lowest_ratio_of_up_service_instances=66 result 0
{"last_sync_time":"201202131117","lowest_ratio_of_up_service_instances":"66"}
got all service result result=0
del meta attr last_sync_time result=0
del meta attr lowest_ratio_of_up_service_instances result=0
{}
got empty service result result=0
